Source/Diagnostics/BoostedFrameDiagnostic.cpp

```diff
@@ -589,15 +601,79 @@ BoostedFrameDiagnostic(Real zmin_lab, Real zmax_lab, Real v_window_lab,
         // x and y bounds are the same for lab frame and boosted frame
         prob_domain_lab.setLo(AMREX_SPACEDIM-1, zmin_lab + v_window_lab * t_lab);
         prob_domain_lab.setHi(AMREX_SPACEDIM-1, zmax_lab + v_window_lab * t_lab);
-        // Construct LabSnapShot
-        LabSnapShot snapshot(t_lab, t_boost, prob_domain_lab,
-                             prob_ncells_lab, ncomp_to_dump,
-                             mesh_field_names, i, *this);
-        snapshots_.push_back(snapshot);
-        buff_counter_.push_back(0);
-        if (WarpX::do_boosted_frame_fields) data_buffer_[i].reset( nullptr );
+        Box diag_box = geom.Domain();
+        LabFrameDiags_[i].reset(new LabFrameSnapShot(t_lab, t_boost,
+                                inv_gamma_boost_, inv_beta_boost_, dz_lab_,
+                                prob_domain_lab, prob_ncells_lab,
+                                ncomp_to_dump, mesh_field_names, prob_domain_lab,
+                                diag_box, i));
     }
 
+    amrex::Real zmin_slice_lab;
+    amrex::Real zmax_slice_lab;
+    amrex::Real cell_dx = 0;
+    amrex::Real cell_dy = 0;
+    IntVect slice_ncells_lab ;
+    Box slicediag_box;
+
+    if ( N_slice_snapshots_ > 0) {
```

Comment:
Could you let us know what you are doing here, if ( N_slice_snapshots_ > 0)?

Reply:
I have added some description above the code line describing that if reduced diagnostics is used then we obtain the Box for the diag using the lo and hi and the corresponding indices of the lab-frame 'slice' (Which can be 1D, 2D, or 3D).